### DISCUSSION TITLE
Vscodium - snap or flatpak - can't run javascript.

### DISCUSSION THREAD

**Discussion by wisdomlight** [VSCodium/vscodium#2081] [Q&A]
on two seprate installation flatpak and snap I couldn't run a javascript from vscodium.
node and npm are both installed on my system - v23.1.0 and 10.9.0 respectivly
I can run the index.js from the system terminal but not from vscodium.
in defaultSettings.json 
terminal.integrated.env.linux: {}  
I am not permitted to add the path to node - it says editor is in read mode only because the file system of the file is read-only
any advice

**Comment by Jeong-ilKajokaya** [comment]
Yo @wisdomlight  👋

I feel you — this is a classic sandbox / path issue when running VSCodium as Flatpak or Snap. Node works fine in your system terminal, but inside VSCodium it’s basically running in a restricted environment and can’t see your system PATH by default.

---

### Why this happens

* Flatpak / Snap apps are sandboxed and don’t inherit your normal environment variables fully. That’s why `node` works in your terminal but VSCodium can’t find it.
* Editing `defaultSettings.json` won’t work because it’s read-only; you’re supposed to use **user settings** instead.

---

### How to fix it

1. **Set the PATH in user settings**

   * Open `Settings` (Ctrl+,) → search for `terminal.integrated.env.linux`
   * Add your Node path there. Example:

     ```json
     "terminal.integrated.env.linux": {
         "PATH": "/usr/bin:/usr/local/bin:/home/youruser/.nvm/versions/node/v23.1.0/bin:$PATH"
     }
     ```
   * Replace with whatever path `which node` gives you.

2. **Use the VSCodium terminal instead of trying to run scripts via the editor**

   * Open integrated terminal (Ctrl+`) and run `node index.js`. The PATH you set above ensures it sees Node.

3. **Alternative: Flatpak / Snap overrides**

   * For Flatpak, you can run:

     ```bash
     flatpak override --user --env=PATH=$PATH com.vscodium.VSCodium
     ```
   * For Snap, you might need to connect classic interfaces or use `snap run --shell vscodium` to launch it with full env.

4. **Consider a non-sandboxed version**

   * If you keep running into PATH / permission issues, using a `.deb` or AppImage version of VSCodium avoids these sandbox restrictions.

---

Basically, the issue isn’t VSCodium itself — it’s Flatpak / Snap restricting Node visibility. Fixing the terminal PATH or using a non-sandboxed build solves it.
